src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttachmentController.php

```diff
@@ -252,6 +266,64 @@ public function deleteBulkAction(Request $request): RedirectResponse
         return $this->redirectToRoute('admin_attachments_index');
     }
 
+    /**
+     * @AdminSecurity("is_granted('read', 'AdminProducts') || is_granted('read', 'AdminAttachments')")
+     *
+     * @param int $attachmentId
+     *
+     * @return JsonResponse
+     */
+    public function getAttachmentInfoAction(int $attachmentId): JsonResponse
+    {
+        $attachmentInfo = $this->getQueryBus()->handle(new GetAttachmentInformation($attachmentId));
+
+        return $this->json(['attachmentInfo' => $this->presentAttachmentInfo($attachmentInfo)]);
+    }
+
+    /**
+     * @AdminSecurity("is_granted('read', 'AdminProducts') || is_granted('read', 'AdminAttachments')")
+     *
+     * @param string $searchPhrase
+     *
+     * @return JsonResponse
+     */
+    public function searchAction(string $searchPhrase): JsonResponse
+    {
+        try {
+            /** @var AttachmentInformation[] $attachments */
+            $attachments = $this->getCommandBus()->handle(new SearchAttachment($searchPhrase));
+        } catch (EmptySearchException $e) {
+            return $this->json(
+                [$e, 'message' => $this->getErrorMessageForException($e, $this->getErrorMessages($e))],
+                Response::HTTP_NOT_FOUND
+            );
+        }
+
+        $result = [];
+        foreach ($attachments as $attachment) {
+            $result[] = $this->presentAttachmentInfo($attachment);
+        }
+
+        return $this->json($result);
+    }
+
+    /**
+     * @param AttachmentInformation $productAttachmentInfo
+     *
+     * @return array<string, mixed>
+     */
+    private function presentAttachmentInfo(AttachmentInformation $productAttachmentInfo): array
```

Comment:
Why don't you move it to a Presenter class ?

Reply:
Which is testable :)

Reply:
Mmmm, I think it would be overkill